contrib/CMakeLists.txt

```diff
@@ -185,7 +185,6 @@ if (USE_INTERNAL_PARQUET_LIBRARY)
 
     # The library is large - avoid bloat.
     target_compile_options (${ARROW_LIBRARY} PRIVATE -g0)
-    target_compile_options (${THRIFT_LIBRARY} PRIVATE -g0)
```

Comment:
Because thrift maybe a commonly used third-party lib, so I seperated thrift library from arrow-cmake. See more details in `cmake/find/thrift.cmake` and `contrib/thrift-cmake`